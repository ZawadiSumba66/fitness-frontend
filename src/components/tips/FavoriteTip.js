import PropTypes from 'prop-types';

const FavoriteTip = ({ title, description, handleRemove }) => (
  <div className="col-md-4 my-2">
    <div className="shadow rounded favourites__card mx-auto pl-3">
      <div className="position-relative">
        <h3 className="text-center font-weight-bold test-name">
          {title}
        </h3>
        <p>{description}</p>
        <button
          onClick={handleRemove}
          type="button"
          className="button-orange font-weight-bold text-light btn"
        >
          Remove
        </button>
      </div>
      <div className="d-flex justify-content-between align-items-center p-3">
        <div>
          <span className="fa fa-star checked" />
          <span className="fa fa-star checked" />
          <span className="fa fa-star checked" />
          <span className="fa fa-star checked" />
          <span className="fa fa-star text-sm-gray" />
        </div>
      </div>
    </div>
  </div>
);

FavoriteTip.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  handleRemove: PropTypes.func.isRequired,
};

export default FavoriteTip;
