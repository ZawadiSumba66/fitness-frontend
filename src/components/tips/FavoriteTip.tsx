import { Link } from '@reach/router';

type TipFavorite = {
  title: string,
  description: string,
  id: number,
  handleRemove: () => void;
};

const FavoriteTip = ({
  title, description, handleRemove, id,
}: TipFavorite) => (
  <div className="col-md-4 my-2">
    <div className="shadow rounded favourites__card mx-auto pl-3">
      <div className="position-relative">
        <h3 className="text-center font-weight-bold test-name">
          <Link to={`/tips/${id}`}>
            {' '}
            {title}
          </Link>
        </h3>
        <p>{description}</p>
        <button
          onClick={handleRemove}
          type="button"
          className="button-orange font-weight-bold text-light btn mt-2"
        >
          Remove
        </button>
      </div>
      <div className="d-flex justify-content-between align-items-center p-3">
        <div>
          <span className="fa fa-star checked" />
          <span className="fa fa-star checked" />
          <span className="fa fa-star checked" />
          <span className="fa fa-star checked" />
          <span className="fa fa-star text-sm-gray" />
        </div>
      </div>
    </div>
  </div>
);

export default FavoriteTip;
