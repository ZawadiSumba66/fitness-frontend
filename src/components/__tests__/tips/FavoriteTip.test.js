import renderer from 'react-test-renderer';
import FavoriteTip from '../../tips/FavoriteTip';

it('matches the favorite tips snapshot', () => {
  const tree = renderer.create(<FavoriteTip />).toJSON();
  expect(tree).toMatchSnapshot();
});
