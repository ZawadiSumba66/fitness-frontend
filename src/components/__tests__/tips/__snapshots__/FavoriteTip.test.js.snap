// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the favorite tips snapshot 1`] = `
<div
  className="col-md-4 my-2"
>
  <div
    className="shadow rounded favourites__card mx-auto pl-3"
  >
    <div
      className="position-relative"
    >
      <h3
        className="text-center font-weight-bold test-name"
      >
        <a
          href="/tips/undefined"
          onClick={[Function]}
        >
           
        </a>
      </h3>
      <p />
      <button
        className="button-orange font-weight-bold text-light btn mt-2"
        type="button"
      >
        Remove
      </button>
    </div>
    <div
      className="d-flex justify-content-between align-items-center p-3"
    >
      <div>
        <span
          className="fa fa-star checked"
        />
        <span
          className="fa fa-star checked"
        />
        <span
          className="fa fa-star checked"
        />
        <span
          className="fa fa-star checked"
        />
        <span
          className="fa fa-star text-sm-gray"
        />
      </div>
    </div>
  </div>
</div>
`;
